docs/pages/versions/unversioned/guides/push-notifications.md

```diff
@@ -283,6 +283,92 @@ type PushMessage = {
    */
   ttl?: number,
 
+  /**
+   * Rich content that accomplishes the push notification.
+   * Note that for iOS, the displaying priority is video > audio > image.
+   * In other words, if the message specifies both `video` and `image`,
+   * the video will be displayed on iOS devices (and the image will be
+   * displayed on Android devices).
+   */
+  richContent?: {
+    /**
+     * Remote https url of an image that will be displayed with the notification.
+     * The image should not have an alpha channel.
+     * Image restrictions on iOS: https://developer.apple.com/documentation/usernotifications/unnotificationattachment.
+     * Image formats supported on Android: JPEG, PNG, and GIF (will not be animated).
+     */
+    image?: string | {
+      url: string,
+      options?: {
+        /**
+         * Whether the image's thumbnail will be displayed.
+         * Defaults to `false`.
+         */
+        thumbnailHidden?: boolean,
+
+        /**
+         * (iOS-specific field)
+         * The clipping rectangle for a thumbnail image. Each value in this key
+         * is a dictionary containing a unit rectangle whose values are in the
+         * range 0.0 to 1.0 and represent the portion of the original image that
+         * you want to display.
+         * For example, specifying `x: 0.25, y: 0.25, width: 0.5, height: 0.5`
+         * defines a clipping rectangle that shows only the center portion of
+         * the image.
+         * Learn more: https://developer.apple.com/documentation/usernotifications/unnotificationattachmentoptionsthumbnailclippingrectkey
+         */
+        thumbnailClippingRect?: {
+          x: number,
+          y: number,
+          width: number,
+          height: number
+        }
+      }
+    },
+
+    /**
+     * (iOS-specific field)
+     * Remote https url of an audio that will be displayed with the notification.
+     * Audio restrictions: https://developer.apple.com/documentation/usernotifications/unnotificationattachment
+     */
+    audio?: string,
+
+    /**
+     * (iOS-specific field)
+     * Remote https url of a video that will be displayed with the notification.
+     * Video restrictions: https://developer.apple.com/documentation/usernotifications/unnotificationattachment
+     */
+    video?: string | {
+      url: string,
+      options?: {
+        /**
+         * Whether the image's thumbnail will be displayed.
+         * Defaults to `false`.
+         */
+        thumbnailHidden?: boolean,
+
+        /**
+         * The clipping rectangle for a thumbnail image. Refer to the option
+         * `image.options.thumbnailClippingRect` above.
+         */
+        thumbnailClippingRect?: {
+          x: number,
+          y: number,
+          width: number,
+          height: number
+        },
+
+        /**
+         * For a video, it is the time (in seconds) into the video from which to
+         * grab the thumbnail image. For an animated image (i.e. a GIF file),
```

Comment:
GIF would be a `video`. This is because Android will display GIFs as a static image, and to make clear of that, it would make more sense to only refer to static images as `image` and both GIFs and videos as `video`.

Reply:
hmm, okay -- I see your reasoning, but that seems a little bit unintuitive to me since in almost all other contexts we are used to thinking of GIF files as images rather than videos. I think many people will naturally want to put GIFs in `image` (esp. if they aren't reading the documentation carefully), and so I think it might be better to expect GIFs in `image` and just document the cross-platform differences in the behavior. What do you think about that?

Reply:
(fwiw I don't think this changes your implementation at all, other than the schema validation on the server)

Reply:
I think it might be confusing for an image to have a thumbnail time property. Also some developers might want to use an animated GIF on iOS and a static image on Android (since animated GIF won't work), and this way `video` nicely override `image` on iOS. Maybe we could add a note in front of `image` to denote that an (animated) GIF should be added as a `video`?

Reply:
I added some documents above.
```
      * (Note that an animated GIF will not be animated on Android devices. If you
     * wish to use an animated GIF for iOS and a static image for Android, put the
     * GIF as a `video` (see below) and the static image as an `image`.)
```

Also, I think having a platform key might be unnecessary. Android only supports image, and iOS only supports displaying one type of media at a time. So the user should just specify the content they need.

My concern is still that it feels unintuitive for an `image` to have `thumbnailTime`. Also, on iOS, if you state `thumbnailTime` for a non-animated image, the image will disappear altogether which might be confusing for people.